mantidproject/mantid
# ReflectometryReductionOneAuto code tidied

**Description of work.**
- `RebinParams` class simplified
- Replaced `catch (...)` with check to see if workspace exists
- Made west const consistent
- Refactored duplicated transmission run code in `validateInputs` into a function
- Added check to prevent unhandled exception when comparing informative/input workspace names

**To test:**
Code review.

We should also do some manual testing as this PR makes changes to functional code relating to re-binning, transmission run validation and slicing.

1. - Open `ISIS Reflectometry` interface
2. - Change instrument to `OFFSPEC`
3. - Enter run number `44956`, angle `0.4` and first trans run `44937`
4. - Process and plot
5. - Alter `QMin`/`QMax`/`dQ/Q` values
6. - Process and plot again

- Delete all workspaces

1. - Change instrument to `INTER`
2. - Enter run number `13460` and angle `0.5`
3. - Click on `Event Handling` tab
4. - Enable `Uniform slicing`
5. - Increase number of slices to `2`
6. - Process and observe output workspaces are correct


Closes #26499 

*This does not require release notes as there is no change in functionality*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
